csrc/ir/utils.h

```diff
@@ -66,13 +66,26 @@ struct MatmulInputs {
   Val* bias = nullptr;
   Val* alpha = nullptr;
   Val* beta = nullptr;
+  // Ordering of dimensions M,N,K in inputs.
+  // Determined based on position of iterdomains.
```

Comment:
Why is `n_pos` or `k_pos` 2? For example, in `M,K x N,K`, `K` is dim 1 in lhs and again dim 1 in rhs. Does `k_pos=2` mean anything? Or is it an arbitrary way to distinguish `[K,N]` and `[N,K]` so `k_pos=10` should work equally well? 

To complicate this even more, while I understand we don't support `K,M x N,K` at this moment, how do you expect the following notation to be extended for that? 

To be more constructive, it **may** be less ambiguous and more extensible if we split k_pos into lhs_k_pos and rhs_k_pos, because k is a contracting dimension that appears in both operands. Does that make any sense?

Reply:
Resolved in an offline discussion.